File: spring-di-quickstart/src/test/java/org/acme/spring/di/GreetingResourceTest.java
package org.acme.spring.di;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.is;

import org.junit.jupiter.api.Test;

import io.quarkus.test.junit.QuarkusTest;

@QuarkusTest
public class GreetingResourceTest {

    @Test
    public void testHelloEndpoint() {
        given()
                .when().get("/greeting")
                .then()
                .statusCode(200)
                .body(is("HELLO WORLD!"));
    }

}

File: application-configuration-kotlin/src/test/kotlin/org/acme/config/GreetingResourceTest.kt
package org.acme.config

import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured.given
import org.hamcrest.CoreMatchers.`is`
import org.junit.jupiter.api.Test

@QuarkusTest
open class GreetingResourceTest {
    @Test
    fun testHelloEndpoint() {
        given()
            .`when`().get("/greeting")
            .then()
            .statusCode(200)
            .body(`is`("hello quarkus!"))
    }
}